Client/Frontend/Browser/Onboarding/Welcome/WelcomeViewCallout.swift

```diff
@@ -0,0 +1,478 @@
+// Copyright 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import UIKit
+import SnapKit
+import BraveUI
+import Shared
+import pop
+
+enum WelcomeViewCalloutState {
+    case welcome(title: String)
+    case privacy(title: String, details: String, buttonTitle: String, action: () -> Void)
+    case defaultBrowser(title: String, details: String, primaryButtonTitle: String, secondaryButtonTitle: String, primaryAction: () -> Void, secondaryAction: () -> Void)
+    case defaultBrowserWarning(title: String, details: String, primaryButtonTitle: String, secondaryButtonTitle: String, primaryAction: () -> Void, secondaryAction: () -> Void)
```

Comment:
These enum cases are a bit large, 3 is usually the max I'd go, can we make a struct and use that instead as the associated value in these cases?

Reply:
Changed at https://github.com/brave/brave-ios/pull/4508/commits/666c9b3796b4fe4de2c9304c8ba8dd99b599a8ec